### DISCUSSION TITLE
An option to generate a Dockerile to make this more developer centric.

### DISCUSSION THREAD

**Discussion by nithinbandaru1** [dotnet/sdk-container-builds#538] [Ideas]
I see that the tool is super great and very helpful, but many developers love seeing the docker files. It give a feeling of having more control over their applications.

It would be great if there is an option to generate docker-file from visual studio.
There can also be a generated docker-file added into projects on build as many other tools does.

I also think that it will not be a feature that would need some maintenance in future once implemented because this will be just providing a view or adding the contents to the file at the end.

Showing artifacts (files) that are generated is better than totally hiding them from developers. It just gives a better feeling as I said.

**Comment by baronfel** [comment]
Hi - this is tracked in https://github.com/dotnet/sdk-container-builds/issues/146, and while not a current priority for the team, I definitely agree that it would be useful to have. We'd love to accept a PR for this functionality!

**Reply by nithinbandaru1** [reply]
Aa great. Thanks for the super-fast response. 

Will have a look at the code repo if I can contribute.
Would be pretty new for me to see how build tasks, file generations work in visual studio but for sure I will put some effort into understanding the library.

**Reply by baronfel** [reply]
Awesome! Note that the actual code for the container tools is over in the https://github.com/dotnet/sdk repo, in the src/Containers subdirectory. Feel free to ping me or the @dotnet/sdk-container-builds-maintainers team with any questions of suggestions for the contribution guide!

**Comment by baronfel** [comment]
In the meantime, the Docker tools for both Visual Studio and VS Code can generated a Dockerfile for your project that works in many cases.
